### PR TITLE
Fix double reincarnation

### PR DESCRIPTION
**Pull request by thanodnl**
This PR adds an integration tests assessing that reincarnation only happens for changes that would actually override the state of the node and don't reincarnate if the change will not result in an overridden state.

### CODE REVIEW

File: test/double-reincarnation.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+test2('ringpop should not bump incarnation number when gossip is older', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) { return [
+        // do not disable node
+        dsl.sendPing(t, tc, 0, {
+            sourceIx: 0,
+            subjectIx: 'sut',
+            status: 'suspect',
+            subjectIncNoDelta: -1 // send a gossip with an older incarnation number
+        }),
+        dsl.waitForPingResponse(t, tc, 0, 1, true),
```
Comment (mennopruijssers) [inline test/double-reincarnation.js]: Why the extra 2 arguments (`1, true`)? `waitForPingResponse` only expects three? (https://github.com/uber/ringpop-common/blob/5cc7e1dcdf5fa7feb8b8b625f15ac22b5ce781f9/test/ringpop-assert.js#L269)

**Comment by mennopruijssers** [comment]
one question but: LGTM!

File: test/double-reincarnation.js (hunk 1 above)
Comment (thanodnl) [inline test/double-reincarnation.js]: Ah, that is strange. I cheated a bit and took the base of this test from here: https://github.com/uber/ringpop-common/blob/fix/double-reincarnation/test/incarnation-no-tests.js#L34

I will remove it in this test 😃

File: test/ringpop-assert.js [hunk 2]
```diff
@@ -896,7 +896,7 @@ function piggyback(tc, opts) {
 
     if(opts.subjectIx === 'sut') {
         update.address = tc.sutHostPort;
-        update.sourceIncarnationNumber = tc.test_state['sutIncarnationNumber'];
```
Comment (Motiejus) [inline test/ringpop-assert.js]: Can you help me understand why this change is needed?

**Comment by Motiejus** [comment]
LGTM for the test, but see my question about the last line of the patch.

File: test/ringpop-assert.js (hunk 2 above)
Comment (thanodnl) [inline test/ringpop-assert.js]: It is hard to see without more context, but the `piggyback` function consists of roughly two parts. The first part is about the source of the gossip, the second part is about the subject. We are here in the part that is about the subject.

Even though we are setting the subject information this line had a bug where the incarnation number of the source of the gossip was set to the incarnation number of the sut. If you look at a bit more context below this line you will see that all branches set the `incarnationNumber` except for this branch.

Comment (Motiejus) [inline test/ringpop-assert.js]: That makes sense. Thank you!

**Comment by Motiejus** [comment]
LGTM
